src/lib/rules/http-cache/http-cache.ts

```diff
@@ -0,0 +1,423 @@
+/**
+ * @fileoverview Checks if your cache-control header and asset strategy follows best practices
+ */
+import * as pluralize from 'pluralize';
+
+import { Category } from '../../enums/category';
+import { RuleContext } from '../../rule-context';
+import { IRule, IRuleBuilder, IFetchEnd } from '../../types';
+
+type targetType = 'fetch' | 'manifest' | 'target';
+type Directives = Map<string, number>;
+type ParsedDirectives = {
+    header: string;
+    invalidDirectives: Map<string, string>;
+    invalidValues: Map<string, string>;
+    usedDirectives: Directives;
+};
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** Max time the HTML of a page can be cached.*/
+        const maxAgeTarget = context.ruleOptions && context.ruleOptions.maxAgeTarget || 180; // 3 minutes
+        /** Max time a resource (CSS, JS, images, etc.) can be cached.*/
+        const maxAgeResource = context.ruleOptions && context.ruleOptions.maxAgeResource || 31536000; // 1 year
+        /** Resources' mediaType that should be cached for a long time.*/
+        const longCached = [
+            'application/manifest+json',
+
+            'audio/ogg',
+            'audio/mpeg',
+            'audio/mp4',
+
+            'font/collection',
+            'font/eot',
+            'font/opentype',
+            'font/otf',
+            'font/ttf',
+            'font/woff',
+            'font/woff2',
+
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/svg+xml',
+            'image/webp',
+            'image/x-icon',
+
+            'text/css',
+            'text/javascript',
+
+            'video/mp4',
+            'video/ogg',
+            'video/webm'];
+        /** The predefined patterns for file revving.*/
+        const predefinedRevvingPatterns: Array<RegExp> = [
+            /\w+\.\w+\.\w{1,3}$/i,
```

Comment:
Can you provide a bit of explanation here?

Also why `{1,3}`? File extensions can be even longer (e.g. `webmanifest`).

Reply:
Fixed